packages/fether-react/src/utils/withTokens.js

```diff
@@ -0,0 +1,71 @@
+import { chainName$ } from '@parity/light.js';
+import { combineLatest } from 'rxjs';
+import { compose, mapPropsStream, withProps, withHandlers } from 'recompose';
+import light from '@parity/light.js-react';
+import localForage from 'localforage';
+import LS_PREFIX from '../stores/utils/lsPrefix';
+import { map, switchMap } from 'rxjs/operators';
+
+import ethereumIcon from '../assets/img/tokens/ethereum.png';
+import localForage$ from './localForage';
+import withAccount from './withAccount';
+
+const LS_KEY = `${LS_PREFIX}::tokens`;
+
+const DEFAULT_TOKENS = {
+  ETH: {
+    address: 'ETH',
+    decimals: 18,
+    logo: ethereumIcon,
+    name: 'Ether',
+    symbol: 'ETH'
+  }
+};
+
+// We have one key per chain per account, in this format:
+// __paritylight::tokens::0x123::kovan
+const getLsKey = ({ accountAddress, chainName }) =>
+  `${LS_KEY}::${accountAddress}::${chainName}`;
+
+/**
+ * HOC which injects the user's whitelisted tokens (stored in localStorage).
+ */
+const withTokens = compose(
+  // Inject chainName and accountAddress into props
+  light({
+    chainName: () => chainName$({ withoutLoading: true })
+  }),
+  withAccount,
+  // mapPropsStream and add localForage$
+  mapPropsStream(props$ =>
```

Comment:
https://github.com/paritytech/fether/blob/f08cf81a88d721d8b65542a9b0c3116322d6b869/packages/fether-react/src/utils/withTokens.js#L40-L47

This can be rewritten as

```js
  mapPropsStream(
    switchMap(props =>
      localForage$(getLsKey(props))
        .pipe(
          map(tokens => ({ ...props, tokens: tokens || DEFAULT_TOKENS }))
        )
    )
  ),
```